later/lib/domain/repository/record_repository_imlp.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:injectable/injectable.dart';
+import 'package:later/database/datasources/records_data_source.dart';
+import 'package:later/domain/models/record_model.dart';
+import 'package:later/domain/remap/record_remapper.dart';
+import 'package:later/domain/repository/record_repository.dart';
+
+@lazySingleton
```

Comment:
We must tell the `injectable` that `RecordRepositoryImpl` must be registered as `RecordsRepository`. When you simply mark a class with `@lazySingleton`, `injectable` registers it as the type of class.

What we want is:

```dart
final recordRepository = injector<RecordsRepository>();
```

But in the current state it will only work like this:
```dart
final recordRepository = injector<RecordsRepositoryImpl>(); // <— Impl!
```